### DISCUSSION TITLE
Does caching CosmosClient affect TokenCredential behavior?

### DISCUSSION THREAD

**Discussion by levimatheri** [Azure/azure-cosmos-dotnet-v3#2842] [Q&A] [answered]
1. My understanding is that the `TokenCredential` handles refreshing tokens when they expire. This is correct, right? 
2. We've been caching the `CosmosClient` in-memory per application. Now we are starting to use Managed identities to access Cosmos, would caching the `CosmosClient` affect token auto-refreshing?

**Comment by levimatheri** [comment, marked as answer]
I decided to test this myself and the background token refresh loop DOES refresh the token in the cache.

**Reply by j82w** [reply]
@levimatheri is correct. The SDK does have a background task to refresh the token to ensure it is always valid. Otherwise all the requests would be blocked until a new token is retrieved which would add unnecessary latency to all the requests.
